pyproject.toml

```
@@ -0,0 +1,66 @@
+[build-system]
+requires = ["setuptools>=62.6", "setuptools_scm[toml]>=6.2"]
```

Currently we are using `versioneer`. If we stick to `versioneer`, we would want to add it here

```suggestion
requires = ["setuptools>=62.6", "versioneer[toml]>=0.28"]
```

Then we would want to follow [these steps]( https://github.com/python-versioneer/python-versioneer#build-time-dependency-mode )

<hr>

The alternative is we could replace `versioneer` with `setuptools_scm`, which would mean delete this file and any reference to it (might need to `grep versioneer` to find anything else)

https://github.com/dask/dask-image/blob/264bd1d842cb6e4c8df24492ed303c32ab7222e1/versioneer.py#L4
https://github.com/dask/dask-image/blob/264bd1d842cb6e4c8df24492ed303c32ab7222e1/MANIFEST.in#L13

<hr>

No strong feelings on either approach. Mainly trying to give more context on the issues encountered and options available

---

Ugh, somehow I thought dask-image had changed from versioneer to setuptools-scm. I don't know why I had that idea - will have to go and untangle some of the mess I've made here

---

Think I hadn't been aware of this discussion and just updated `versioneer` in https://github.com/dask/dask-image/pull/357. However, `setuptools_scm` seems much lighter and simpler to maintain.

---

When looking for occurrences of "setup.py" I found that versioneer.py contains many of them (in the comments). Maybe another reason to switch to `setuptools_scm`?

---

No strong feelings between either. They both work well from experience

---

Good to know! In this case, in the interest of advancing this PR we could handle a potential switch in a separate PR (since here everything seems to be compatible with versioneer).